tutorials/tts/TTS_pronounciation_customization.ipynb

```diff
@@ -0,0 +1,290 @@
+    "* *known* words - words that are present in the model's phonetic dictionary\n",
+    "* *out-of-vocabulary (OOV)* words - words that are missing from the model's phonetic dictionary. \n",
+    "* *[heteronyms](https://en.wikipedia.org/wiki/Heteronym_(linguistics)* - words with the same spelling but different pronunciations and/or meanings, e.g., *bass* (the fish) and *bass* (the musical instrument).\n",
+    "\n",
+    "#### Important NeMo flags:\n",
+    "* `your_spec_generator_model.vocab.g2p.phoneme_dict` - model's phonetic dictionary that maps words to their phonetic form, e.g., [ARPABET-based CMU Dictionary](https://github.com/NVIDIA/NeMo/blob/r1.14.0/scripts/tts_dataset_files/cmudict-0.7b_nv22.10) or [IPA-based CMU Dictionary](https://github.com/NVIDIA/NeMo/blob/r1.14.0/scripts/tts_dataset_files/ipa_cmudict-0.7b_nv22.10.txt)\n",
+    "* `your_spec_generator_model.vocab.g2p.ignore_ambiguous_words`: if is set to **True**, words with more than one phonetic representation in the pronunciation dictionary are ignored. This flag is relevant to heteronyms and non-heteronym words with multiple valid phonetic forms in the dictionary, for example, due to accent variations.\n",
+    "\n",
+    "TTS models take text in grapheme form, then convert all known unambiguous words into phonetic form during preprocessing. The rest of the words (OOV and words with multiple dictionary entries) are kept as graphemes, and the TTS model uses context clues from the sentence to predict an appropriate pronunciation for such words.\n",
+    "\n",
+    "To ensure the desired pronunciation, we need to add a new entry to the models' phonetic dictionary. If the target word is already in the dictionary, we need to remove the default pronunciation so that only the target pronunciation is present. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Default G2P\n",
+    "\n",
+    "Below we show how to analyze default G2P output of the NeMo models"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import nemo.collections.tts as nemo_tts\n",
+    "from nemo_text_processing.g2p.modules import IPAG2P\n",
+    "import soundfile as sf\n",
+    "import IPython.display as ipd\n",
+    "import torch\n",
+    "\n",
+    "# Load mel spectrogram generator\n",
+    "spec_generator = nemo_tts.models.FastPitchModel.from_pretrained(\"tts_en_fastpitch_ipa\").eval()\n",
+    "# Load vocoder\n",
+    "vocoder = nemo_tts.models.HifiGanModel.from_pretrained(model_name=\"tts_hifigan\").eval()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# to use dictionary entries for known words\n",
+    "spec_generator.vocab.phoneme_probability = 1\n",
+    "spec_generator.vocab.g2p.phoneme_probability = 1\n",
```

Comment:
I think these shouldn't be necessary based on this line in `parse()`: https://github.com/NVIDIA/NeMo/blob/main/nemo/collections/tts/models/fastpitch.py#L268

Since the newer checkpoint has `learn_alignment=True`, this sets phoneme probability to 1 automatically during the parsing (L271-272) even though printing phoneme probability is 0.8 normally.